client/extensions/woocommerce/app/products/product-variation-types-form.js

```diff
@@ -31,13 +31,19 @@ export default class ProductVariationTypesForm extends Component {
 		if ( ! product.attributes ) {
 			this.addType();
 		}
+
+		this.debouncedUpdateName = debounce( this.updateName, 300 );
 	}
 
 	constructor( props ) {
 		super( props );
 
+		this.state = {
+			attributeNames: [],
```

Comment:
I didn't have a specific reason, so I've switched this to an object to be a little clearer. You can still easily clone with the spread operator.